### DISCUSSION TITLE
Climate control toggle not switching icons

### DISCUSSION THREAD

**Discussion by franchiceddhu** [cgiesche/streamdeck-homeassistant#280] [Help and Support]
Hello, I've been trying to create a switch for climate control in my room. I've correctly setup up a "toggle" action on short press of the button, and set a custom title like this: {{temperature}}°C {{state}}.

Everything works perfectly, yet when I actually press the button, it won't cycle between the two icons I set for "off" and "on" state. It will stay fixed on OFF state icon, even if my A/C unit switches on or off using the button. It is worth noting that the {{state}} variable in my custom title shows "cool" or "off" correctly.

Also, I've already set up a similar button for a light switch, and it worked without problems.

Thanks for you great extension!

**Comment by GoBoKri** [comment]
Have the exact same issue. Guess it only triggers on "on" and not "cool"?
